packages/sil-governance/src/routes/dashboard/index.e2e.spec.ts

```diff
@@ -0,0 +1,132 @@
+import express, { Request, Response } from "express";
+import { Server } from "http";
+import { Browser, Page } from "puppeteer";
+import { sleep } from "ui-logic";
+
+import { TX_CONFIRMATION_VIEW_ID } from "../../components/ConfirmTransaction";
+import { closeBrowser, createExtensionPage, createPage, launchBrowser } from "../../utils/test/e2e";
+import { whenOnNavigatedToE2eRoute } from "../../utils/test/navigation";
+import { acceptEnqueuedRequest, loginAsGovernor, rejectEnqueuedRequest } from "../../utils/test/persona";
+import { findRenderedE2EComponentWithId } from "../../utils/test/reactElemFinder";
+import { withChainsDescribe } from "../../utils/test/testExecutor";
+import { CREATE_PROPOSAL_ROUTE, DASHBOARD_ROUTE } from "../paths";
+import {
+  getProposalTitlesE2E,
+  sortByExpiryDateE2E,
+  sortByStartDateE2E,
+  sortByVoteE2E,
+  voteNoOnFirstProposalE2E,
+  voteYesOnFirstProposalE2E,
+} from "./test/operateDashboard";
+import { travelToDashboardE2e } from "./test/travelToDashboard";
+
+withChainsDescribe("E2E > Dashboard route", () => {
+  let browser: Browser;
+  let page: Page;
+  let extensionPage: Page;
+  let server: Server;
+
+  beforeAll(() => {
+    const app = express();
+
+    app.use(express.static(require("path").join(__dirname, "/../../../build")));
+
+    app.get("/*", function(req: Request, res: Response) {
+      res.sendFile(require("path").join(__dirname, "build", "index.html"));
+    });
+
+    server = app.listen(9000);
+  });
+
+  afterEach(async () => {
+    await closeBrowser(browser);
+  });
+
+  afterAll(() => {
+    server.close();
+  });
+
+  it("has an 'Add New Proposal' button that redirects to /create-proposal", async (): Promise<void> => {
+    browser = await launchBrowser();
+    page = await createPage(browser);
+    extensionPage = await createExtensionPage(browser);
+    await loginAsGovernor(page, extensionPage);
+    await sleep(2000);
```

Comment:
Is this possible not to use sleep here? Maybe it is possible to wait for specific part of the page appearance? Like specific ID or element (or maybe this is possible to add ID to wait for it?). If so you can use `page.waitForSelector` instead of sleep.